### ISSUE TITLE
slotted upgrade timing issue causes inputs array to be empty

### ISSUE THREAD

**Issue by robrez** [vaadin/vaadin-custom-field#46]
# Background

When clicking the "label", was getting an error in the console stating:

> vaadin-custom-field-mixin.js:117 Uncaught TypeError: Cannot read property 'focus' of undefined
    at HTMLElement.focus (vaadin-custom-field-mixin.js:117)

```
    focus() {
      this.inputs && this.inputs[0].focus();
    }
```

Found that the inputs array was empty because, i think, the slotted content was not yet upgraded so the filter in `__setInputsFromChildNodes` fails when testing for the existence of `node.validate` or `node.checkValidity`

## Debugging

Modified the function to re-check in in a setTimeout:

```javascript
    __setInputsFromChildNodes() {
      let children = this.querySelectorAll("*");
      console.log("children", children);
      let loop = () => {
        children.forEach(node => {
          console.log("node", node);
          console.log(node.constructor);
          console.log("hasValidate", !!node.validate);
        });
      };
      loop();
      setTimeout(loop, 1000);

      const inputs = Array.from(this.querySelectorAll("*")).filter(
        node => node.validate || node.checkValidity
      );
      this._setInputs(inputs);
      this.__setValue();
    }
```

## Console Output

![image](https://user-images.githubusercontent.com/1993389/64893351-715fdb00-d63c-11e9-8b54-0a7364ca1842.png)

**Comment by Haprog** [comment]
Thanks for the report! Can you provide a sample app or sample code that reproduces this problem case (slotted element not being upgraded before `__setInputsFromChildNodes` is triggered)?

Looks like at least `focus()` method should be updated to check that `inputs` is not an empty array.

**Comment by robrez** [comment]
[Here is a stackblitz](https://stackblitz.com/edit/v-gm9u5o)

Determined that swapping import order seems to be a good workaround (v-custom-field must be imported last for things to work)

**Comment by Haprog** [comment]
Thanks!

**Comment by web-padawan** [comment]
Fixed by #58
